joint_calling/datasource.py

```diff
@@ -0,0 +1,68 @@
+"""This module defines the methods required to authenticate,
+access, filter data from a datasource (e.g. AirTable).
+It will return a list of files to be moved by the upload
+processor"""
+
+import json
+import abc
+from typing import List
+from airtable import Airtable
+from google.cloud import secretmanager
+
+
+class Datasource(metaclass=abc.ABCMeta):
+    """A source of data"""
+
+    @abc.abstractmethod
+    def get_table(self):
```

Comment:
You might consider making this return a new data structure (that you define) with all the fields that you expect. That way below, instead of an `Airtable`, you can map `Airtable -> YourNewTable`. Adding a new Datasource just means adding a new mapping, rather than changing other code that references the Airtable.

Reply:
Thinking wider for other data sources, is it likely you'll need to specify a TableID? Or does your `Datasource` class correspond to exactly one TableID on `__init__`?